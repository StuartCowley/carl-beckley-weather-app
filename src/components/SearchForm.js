import React from "react";
import PropTypes from "prop-types";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);
  const handleKeyPress = (event) => {
    if (event.key === "Enter") {
      onSubmit();
    }
  };

  return (
    <div className="search-form">
      <input
        className="search-form_text-box"
        type="text"
        onChange={handleInputChange}
        onKeyDown={handleKeyPress}
        value={searchText}
        placeholder="Enter city..."
      />
      <button
        className="search-form_button"
        type="submit"
        onClick={onSubmit}
        data-testid="button"
      >
        Search
      </button>
    </div>
  );
}

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
