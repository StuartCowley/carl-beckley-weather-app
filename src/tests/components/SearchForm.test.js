import React from "react";
import { fireEvent, render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form", () => {
  const validProps = {
    searchText: "Manchester",
    setSearchText: jest.fn,
    onSubmit: jest.fn(),
  };

  it("renders the component", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("fires the correct event", () => {
    const { getByTestId } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    const button = getByTestId("button");
    fireEvent.click(button);
    expect(validProps.onSubmit).toHaveBeenCalled();
  });
});
