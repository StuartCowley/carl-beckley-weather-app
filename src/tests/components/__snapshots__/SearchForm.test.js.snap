// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders the component 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-form_text-box"
      placeholder="Enter city..."
      type="text"
      value="Manchester"
    />
    <button
      class="search-form_button"
      data-testid="button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
